assets/js/wc-gateway-ppec-generate-cart.js

```diff
@@ -30,52 +30,67 @@
 			$( '#woo_pp_ec_button_product > *' ).css( 'pointer-events', 'none' );
 		} );
 
+	var variation_valid = true;
+	var fields_valid = true;
+	var update_button = function() {
+		$( '#woo_pp_ec_button_product' ).trigger( ( variation_valid && fields_valid ) ? 'enable' : 'disable' );
+	};
+
+	var validate_form = function() {
+		silent_validation = true;
+		fields_valid = true;
+		form.find( 'select:enabled, input:enabled, textarea:enabled' ).each( function() {
+			fields_valid = fields_valid && this.checkValidity();
```

Comment:
Is there a reason not use `checkValidity()` on the entire form instead of on each of the input elements?

For example, `form.get( 0 ).checkValidity()`?

Reply:
The reason is that I thought `checkValidity` is only for individual fields (that's what's referred to in almost all the documentation). There's a similar [`form.get(0).reportValidity()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/reportValidity) method, but [it's not supported in IE 11](https://caniuse.com/#feat=constraint-validation).

I've tried `form.get( 0 ).checkValidity()` on https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/pull/507/commits/347195ab13863fbc9a536c2b60f433c0b1363e99 and it works. Great catch!

Also, while testing this, I fired up the old IE11 VM and turns out the validation didn't work as it should. Fixed in https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/pull/507/commits/3f76659905474f0fb07d8df031baa9bdc5cba8cf